### PR TITLE
Replace Google Analytics with Segment

### PR DESCRIPTION
**Pull request by DMarby**
In order to be able to query these events better in BigQuery et-al, I changed the app download to be a single event rather then one per platform, and added the platform and app as dimensions.

I also added the category and label properties, which makes events show up correctly in GA when set as a destination in Segment. This means that it's still easy to filter or join on any combinations of app/platform/etc in GA itself.

image
image

Fixes INS-295

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -112,13 +112,12 @@ module.exports = {
...
     {
-      resolve: `gatsby-plugin-google-analytics`,
+      resolve: 'gatsby-plugin-segment-js',
       options: {
-        trackingId: 'UA-8499472-33',
-        head: false,
-        anonymize: true,
-        respectDNT: true,
-        pageTransitionDelay: 0
+        prodKey: 'eqIeHoa1lca59RxSJxVkX4KZcjkCiOxk',
```
Comment (develohpanda) [inline gatsby-config.js:117]: Double checking whether this should be checked in or come from an environment variable during the build

File: src/components/direct-download-button.js [hunk 2]
```diff
@@ -82,9 +82,15 @@ class DirectDownloadButton extends React.Component {
...
-            action: `${downloadInfo.platformName}`,
-            label: message
+            event: `App Download`,
+            properties: {
+              app: `${appName}`,
+              platform: `${downloadInfo.platformName}`,
+
+              // For GA
+              category: `Download (${appName})`,
+              label: message
```
Comment (develohpanda) [inline src/components/direct-download-button.js:92]: Since GA has been removed, are these still required?

**Review by develohpanda — APPROVED** [review]

**Comment by develohpanda** [comment]
Same note with this PR regarding de-duplication 👍

File: src/components/direct-download-button.js (hunk 2 above)
Comment (DMarby) [reply src/components/direct-download-button.js:92]: Yep, we don't use GA directly anymore, but Segment still sends metrics to GA (as well as other destinations), so this is needed to make the events show up correctly there

Comment (develohpanda) [reply src/components/direct-download-button.js:92]: Ah, I see 👍

File: gatsby-config.js (hunk 1 above)
Comment (DMarby) [reply gatsby-config.js:117]: Yep, fine to check in as it's public (it's used by the client-side JS to identify the metrics), and won't change based on environment.

**Review by nijikokun — APPROVED** [review]
🎉 Looks good to me!

**Pull request merged**